### PR TITLE
new min-funding config parameter for private channels 

### PR DESCRIPTION
**Pull request by remyers**
Add the ability to set a different min-funding limit for private and public channels. See issue "Private channels should have their own dedicated 'min-funding-satoshis' setting" [#2077](https://github.com/ACINQ/eclair/issues/2077).

If `channel.private.enable-alt` = `true` then the `channel.private.min-funding-satoshis` value will be used for private channels, otherwise the `channel.min-funding-satoshis` will be used for both public and private channels.

A few questions:

- previous `min-funding-satoshis` tests compared `assert(error.toAscii ===  Error(...).toAscii)` while other tests in `WaitForOpenChannelStateSpec` left off the `.toAscii`. Is there a reason to use the `.toAscii` in the assert here?
- [Issue 2077 mentioned we should document](https://github.com/ACINQ/eclair/issues/2077#issuecomment-981415320) that users should not set unsafely low min-funding limits for private channels. Only if users set their public channels to have higher limits then they might allow private channels with a lower, but still safe, `min-funding-satoshis`. Should I add documentation to that effect in [docs\Configure.md](https://github.com/ACINQ/eclair/blob/master/docs/Configure.md) ? or in the [release notes](https://github.com/ACINQ/eclair/blob/master/docs/release-notes/eclair-vnext.md) ?

### CODE REVIEW

**Comment by remyers** [comment]
My rational for the `channel.private.enable-alt` is that when people upgrade their node it should default to have the same min-funding behavior as before. If we just added a new setting for private channels then it would use whatever default was in the reference.conf and that would potentially be very different than what the user expects their private channels to use.

I'm not completely happy with the `enable-alt` as the name for that parameters - open to suggestions.

**Comment by pm47** [comment]
@thomash-acinq beat me to it, indeed it's the way to go:
https://github.com/ACINQ/eclair/blob/f3c6c7885de5b4c8c48302b06b25b1e7422c763c/eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala#L173

**Comment by remyers** [comment]
> I understand your rationale for `channel.private.enable-alt` but I really don't like the result. An alternative would be to rename `min-funding-satoshis` to `min-public-funding-satoshis` and add a deprecation message for `min-funding-satoshis`, that way everyone using this parameter will see the message and will be forced to manually change it.

I agree. As long as we mention both new options in the deprecation message that should ensure they know to set both the public and private min funding values.

File: eclair-core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -82,7 +82,8 @@ eclair {
 
     reserve-to-funding-ratio = 0.01 // recommended by BOLT #2
     max-reserve-to-funding-ratio = 0.05 // channel reserve can't be more than 5% of the funding amount (recommended: 1%)
-    min-funding-satoshis = 100000
+    min-public-funding-satoshis = 100000 // high public minimums are attractive to pathfinding heuristics
+    min-private-funding-satoshis = 100000 // safe private minimums increase the cost of spamming and flood-and-loot attacks
```
Comment (thomash-acinq) [inline eclair-core/src/main/resources/reference.conf:86]: I don't think these comments are really helpful. Larger channels are more attractive but that doesn't have anything to do with the minimum we set here, a small channel is more attractive than no channel.

Comment (remyers) [reply eclair-core/src/main/resources/reference.conf]: Removed them in [6534977](https://github.com/ACINQ/eclair/pull/2203/commits/6534977f9f6d81813c31190663da069d230c0343). Sounds like its too nuanced to explain in comments.

**Review by thomash-acinq — APPROVED** [review]

**Comment by thomash-acinq** [comment]
I've just remembered about the release notes, can you update them too?

**Comment by pm47** [comment]
Also, for PR names and squashed commit messages please use a short sentence starting by a verb with an uppercase.

**Pull request merged**